### PR TITLE
(maint) Acceptance - refactor alt helper methods

### PR DESCRIPTION
**Pull request by johnduarte**
This commit removes the acceptance helper methods for building
the alt file paths for pxp-agent certs. It replaces these
methods with an additional boolean parameter on the default
file path methods. This parameter will instruct the method
whether to return the standard or alt paths based on its value.

[skip ci]

### CODE REVIEW

File: acceptance/lib/pxp-agent/config_helper.rb [hunk 1]
```diff
@@ -119,89 +118,55 @@ def pxp_agent_config_file(host)
 
 # @param host a beaker host (to determine the correct path for the OS)
 # @param client_number which client 01-05 you want the key file for
+# @param [boolean] use_alt_path use alt file path?
```
Comment (james-stocks) [inline acceptance/lib/pxp-agent/config_helper.rb]: I think if someone new comes onto the project; this doc comment might be confusing. Perhaps

# @param [boolean] use_alt_path Use the repo's alternative set of ssl certs? Default false, use the standard set of ssl certs

(Too wordy?)

Comment (johnduarte) [inline acceptance/lib/pxp-agent/config_helper.rb]: Yard will introspect the default from the method definition. Each method only defines a single path, not a set of certs. Maybe the following is more accurate?

    # @param [boolean] use_alt_path Use alternative test ssl file if true, otherwise use standard test ssl file

Comment (james-stocks) [inline acceptance/lib/pxp-agent/config_helper.rb]: Perfect!

**Comment by james-stocks** [comment]
Looks good to me.
If this merges, https://github.com/puppetlabs/pxp-agent/pull/262 may need updated, as the PRs are in the same area

File: acceptance/tests/invalid_ssl_config.rb [hunk 2]
```diff
@@ -21,7 +21,7 @@
 invalid_config_mismatching_keys = {:broker_ws_uri => broker_ws_uri(master),
                                    :ssl_key => ssl_key_file(agent1, 1),
                                    :ssl_ca_cert => ssl_ca_file(agent1),
-                                   :ssl_cert => alt_ssl_cert_file(agent1, 1)}
+                                   :ssl_cert => ssl_cert_file(agent1, 1, use_alt_path=true)}
```
Comment (richardc) [inline acceptance/tests/invalid_ssl_config.rb]: This isn't python, so we don't have semi-named arguments here.  All this did was introduce a temporary variable we ignored.

```
ssl_cert_file(agent1, 1, eat_at_joes=true)
```

has the same effect.

Comment (richardc) [inline acceptance/tests/invalid_ssl_config.rb]: Actually to correct myself, ruby 2.0 did introduce named keyword parameters, so If the intent was to use those the code would need to look more like this:

```
def ssl_cert_file(host, client_number, use_alt_path: false)
  # code here
end

ssl_cert_file(agent1, 1, use_alt_path: true)
```

In that pattern use_alt_path is the name of the parameter.

Comment (richardc) [inline acceptance/tests/invalid_ssl_config.rb]: @james-stocks has raised #264 to correct this.

Comment (richardc) [inline acceptance/tests/invalid_ssl_config.rb]: @james-stocks also raised #265 as a non-ruby 2.0 requiring version to correct this.
